src/components/common/PaginatedTxns/index.tsx

```diff
@@ -34,57 +34,59 @@ const TxPage = ({
 }: {
   pageUrl: string
   useTxns: typeof useTxHistory | typeof useTxQueue
-  onNextPage?: (pageUrl?: string) => void
+  onNextPage?: (pageUrl: string) => void
   isFirstPage: boolean
 }): ReactElement => {
-  const { page, error } = useTxns(pageUrl)
+  const { page, error, loading } = useTxns(pageUrl)
   const [filter] = useTxFilter()
-
   const isQueue = useTxns === useTxQueue
 
-  if (page?.results) {
-    return (
-      <>
-        {/* FIXME: batching will only work for the first page results */}
-        {isFirstPage && (
-          <Box display="flex" flexDirection="column" alignItems="flex-end" mt={['-94px', '-44px']} mb={['60px', 0]}>
-            {isQueue ? <BatchExecuteButton items={page.results} /> : <TxFilterButton />}
-            {filter && <Typography mt={2}>{getResultCount(filter, page)}</Typography>}
-          </Box>
-        )}
-
-        {page.results.length ? <TxList items={page.results} /> : isQueue && <NoQueuedTxns />}
-
-        {onNextPage && page.next && (
-          <Box my={4} textAlign="center">
-            <InfiniteScroll onLoadMore={() => onNextPage(page.next)} />
-          </Box>
-        )}
-      </>
-    )
-  }
+  return (
+    <>
+      {isFirstPage && filter && page && (
+        <Box display="flex" flexDirection="column" alignItems="flex-end" pt={[2, 0]} pb={3}>
+          {getFilterResultCount(filter, page)}
+        </Box>
+      )}
 
-  if (error) {
-    return <ErrorMessage>Error loading transactions</ErrorMessage>
-  }
+      {page && <TxList items={page.results} />}
```

Comment:
Not critical but `TxList` is now being rendered as an empty element even if there are no results. We could still check for `page.results.length > 0` before rendering the list.